packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -479,6 +478,103 @@ void Add(
   }
 }
 
+namespace ColMapFunctors
+{
+  template<typename ByteView, typename GView>
+  struct UnionEntries
+  {
+    UnionEntries(ByteView& entryUnion_, const GView& gids_) : entryUnion(entryUnion_), gids(gids_) {}
+    KOKKOS_INLINE_FUNCTION void operator()(const size_t i) const
+    {
+      entryUnion(gids(i)) = 1;
+    }
+    ByteView& entryUnion;
+    const GView& gids;
+  };
+
+  template<typename ByteView, typename LView>
+  struct EntriesPrefixSum
+  {
+    EntriesPrefixSum(const ByteView& entryUnion_, LView& gtol_) : entryUnion(entryUnion_), gtol(gtol_) {}
+    KOKKOS_INLINE_FUNCTION void operator()(const size_t i, size_t& update, const bool finalPass) const
+    {
+      // exclusive scan
+      if(finalPass)
+      {
+        gtol(i) = update;
+      }
+      if(entryUnion(i))
+      {
+        update++;
+      }
+      if(finalPass)
+      {
+        if(i == entryUnion.dimension_0() - 1)
```

Comment:
How do you get that result back on host?  Do you do a `Kokkos::deep_copy` of a 0-D View, or just UVM it?  For the former, see Tpetra_Details_getEntryOnHost.hpp.